src/SDK/MonitoringApplianceEmulator/Converters/ChartPropertyToChartPropertyControlViewModelConverter.cs

```diff
@@ -0,0 +1,66 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ChartPropertyToChartPropertyControlViewModelConverter.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.Converters
+{
+    using System;
+    using System.Globalization;
+    using System.Windows.Data;
+    using Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.ViewModels;
+    using Microsoft.Azure.Monitoring.SmartDetectors.RuntimeEnvironment.Contracts;
+
+    /// <summary>
+    /// Implementation of <see cref="IValueConverter"/> for converting from a <see cref="ChartAlertProperty"/> value to <see cref="ChartPropertyControlViewModel"/>.
+    /// </summary>
+    public class ChartPropertyToChartPropertyControlViewModelConverter : IValueConverter
+    {
+        #region Implementation of IValueConverter
+
+        /// <summary>
+        /// Converts a value.
+        /// </summary>
+        /// <returns>
+        /// A converted value. If the method returns null, the valid null value is used.
+        /// </returns>
+        /// <param name="value">The value produced by the binding source.</param>
+        /// <param name="targetType">The type of the binding target property.</param>
+        /// <param name="parameter">The converter parameter to use.</param>
+        /// <param name="culture">The culture to use in the converter.</param>
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            // This 'if' clause was added to workaround a known issue in WPF -
+            // DataContext of a 'ListViewItem' is being assigned with '{DisconnectedItem}' object and then a null value whenever its container is closed.
+            if (value == null || value.ToString() == "{DisconnectedItem}")
+            {
+                return value;
+            }
```

Comment:
why not just return value in the if below?

Reply:
because I want to be specific - the converter should return the value only if the given value if null or DisconnectedITem. 

---
In reply to: [231015514](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/17#discussion_r231015514) [](ancestors = 231015514)